include/oneapi/mkl/sparse_blas/detail/sparse_blas_ct.hxx

```diff
@@ -0,0 +1,178 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+// This file is meant to be included in each backend sparse_blas_ct.hpp files
+// Each function calls the implementation from onemkl_sparse_blas_backends.hxx
+
+#ifndef BACKEND
+#error "BACKEND is not defined"
+#endif
+
+inline void init_matrix_handle(backend_selector<backend::BACKEND> selector,
+                               matrix_handle_t *handle) {
+    BACKEND::init_matrix_handle(selector.get_queue(), handle);
+}
```

Comment:
Since oneMKL Interfaces and oneMKL Product are relatively compatible, I see this being easy for us, but for cuSPARSE backend, won't we need an internal representation in this oneMKL Interfaces codebase still for making their backend work? If we have to go for PIMPL pattern for the handle for cuSPARSE, we might as well have one for the oneMKL backend and not have the `queue` be in there. Food for thought!

Reply:
We will still need the queue for runtime dispatching. Here in the compile-time dispatching it is easier to always give the queue as a first argument for consistency.
How we will support this with cuSPARSE will depend on which spec changes we make but I don't think we can plan ahead for this in this PR.

Reply:
Resolving as [oneAPI-spec PR#505](https://github.com/oneapi-src/oneAPI-spec/pull/505) adds in a queue to `init_matrix_handle` in the spec.